.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,138 @@
+name: Build Docker
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [linux/amd64, linux/386, linux/arm/v6, linux/arm/v7, linux/arm64]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v1
+        with:
+          buildx-version: latest
+          qemu-version: latest
+
+      - name: Available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      - name: Login to DockerHub Registry
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+
+      - name: Run Buildx
+        run: |
+          docker buildx build \
+            --platform ${{ matrix.platform }} \
+            --output "type=image,push=false" \
+            --cache-to=type=local,dest=docker-cache/${{ matrix.platform }} ./
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: docker-cache
+          path: docker-cache
+
+  deploy_dev:
+    name: Deploy Dev
+    if: startsWith(github.ref, 'refs/heads/') # Just the branches
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set environment
+        run: |
+          echo "::set-env name=DOCKER_REPO::${{ secrets.DOCKER_USERNAME }}/dungeon-revealer"
```

Comment:
I think we must distinguish between DOCKER_USERNAME and DOCKER_REPOSITORY_NAME?  As the user name for login is different from the organization name.